### DISCUSSION TITLE
Support Matrix URIs

### DISCUSSION THREAD

**Discussion by Saklad5** [community/community#46165] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

The new profile options are welcome, but they currently require HTTPS URI schemes.

Other IANA-registered schemes, such as [Matrix](https://www.iana.org/assignments/uri-schemes/prov/matrix), should be allowed.

**Comment by HarisDotParis** [comment]
An alternative approach is to use matrix.to to enable Matrix support in GitHub profiles.

matrix.to is a website by matrix.org that creates simple and readable links to Matrix profiles/users, forwarding users then to a Matrix client of their choice (be it web client, app or CLI).

The only way to make this work better in GitHub profiles is to enable profile settings to recognise these matrix.to addresses and assign the Matrix logo to them. This should be very easy to implement.

**Reply by HarisDotParis** [reply]
This is an examplary link – and as the website doesn't check whether the user exists, even this link works, so feel free to click: [https://matrix.to/#/@user:example.com](https://matrix.to/#/@user:example.com)

**Reply by Saklad5** [reply]
That relies on a specific server being live, which is not ideal.

matrix.to was made as a stopgap pending IANA registration of the URI scheme. It should be avoided if possible.

**Reply by HarisDotParis** [reply]
Fair point, but my approach and possible other approaches can be implemented together. While the approach I mentioned relies on matrix.to, it is very simple to implement from GitHub's side and works universally with all servers and Matrix clients, including Matrix servers that have no web interface at all.

**Reply by Saklad5** [reply]
This has nothing to do with web interfaces.

**Reply by HarisDotParis** [reply]
> This has nothing to do with web interfaces.

Not directly, but it provides a way to link to one's Matrix profile. It is very easy to extract the Matrix ID from it, and it works in all kinds of environments. And currently, it's the only [official](https://matrix.org/faq/#how-do-i-create-a-link-to-a-matrix-room-or-user%3F) way to link to a user profile.

**Reply by Saklad5** [reply]
You are linking to an outdated page that explicitly says it will be replaced by the URI scheme.

The issue here is that it hinges entirely on a centralized server, which defeats the purpose of Matrix.

**Reply by HarisDotParis** [reply]
The linked official FAQ defines the current state of affairs. Yes, it depends on that centralised service, but it is currently the only working solution, whereas other (and I agree, better) ways to link to one's profile are yet to appear. It would be better to have an okay solution that works now instead of waiting for an undefined amount of time till a better solution appears. And that better solution can be implemented as well once it arises.
